vars/dockerExecute.groovy

```diff
@@ -1,51 +1,66 @@
 import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.ConfigurationLoader
 
 def call(Map parameters = [:], body) {
 
     def STEP_NAME = 'dockerExecute'
     def PLUGIN_ID_DOCKER_WORKFLOW = 'docker-workflow'
 
-    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters){
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
         def dockerImage = parameters.dockerImage ?: ''
         Map dockerEnvVars = parameters.dockerEnvVars ?: [:]
         def dockerOptions = parameters.dockerOptions ?: ''
         Map dockerVolumeBind = parameters.dockerVolumeBind ?: [:]
+        final script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
 
-        if(dockerImage) {
+        if (inKubernetes(script)) {
+            if (env.POD_NAME && hasContainerDefined(script, dockerImage)) {
```

Comment:
I suggest we do the same configuration handling using `ConfigurationHelper` here. and then pass `config` instead of script.

Currently this step only supports parameters and implicitly the k8s configuration.
We should have the same semantics as in other steps.